torch_xla/experimental/xla_sharded_tensor.py

```diff
@@ -109,8 +109,7 @@ def __new__(cls, elem: torch.Tensor, *args, **kwargs):
   # which results from the sharding.
   @property
   def local_shards(self) -> List[XLAShard]:
-    shards = torch_xla._XLAC._get_local_shards(self.global_tensor)
-    devices = [str(shard.device) for shard in shards]
+    shards, devices = torch_xla._XLAC._get_local_shards(self.global_tensor)
     indices = torch_xla._XLAC._get_local_shard_indices(self.global_tensor)
     return [
         XLAShard(s.cpu(), i, d) for s, i, d in zip(shards, indices, devices)
```

Comment:
If `shards` are on CPU when returned from `_get_local_shards`, we can remove the `.cpu()` call on `s`

Reply:
I actually don't remember, I can check.

Reply:
it is on CPU already, so I removed the `.cpu` below